server/server.go

```diff
@@ -510,17 +510,26 @@ func (s *Server) startMonitoring(secure bool) error {
 		hp           string
 		err          error
 		httpListener net.Listener
+		port   int
```

Comment:
Should not make a difference. Your change just says that if port is -1, then pick a random port. Since default options are either 0 or set, this change should not make any difference, except for the 2 new tests that you have added. If the port picked for monitor port conflicts with another test (in different package) for either the client, route, or monitor port.